discussion_agents/cog/agent/react.py

```diff
@@ -80,10 +103,20 @@ def generate(self, question: str, reset: bool = True) -> str:
         Args:
             question (str): The question to be processed.
             reset (bool, optional): Whether to reset the internal state before processing. Defaults to True.
-
+            examples (str): The example of text generated.
+            env (Alfworld_environment): The variable to interact with Alfworld.
         Returns:
             str: The accumulated output from the ReAct process.
         """
+
+        benchmark_type = check_type(examples=examples)
```

Comment:
is there a way to leave the benchmark decision solely up to the user? the user should only need to pass some arguments to the react agent to have it run on a benchmark. the agent shouldn't have any knowledge of benchmark types